frontend/src/lib/api.ts

```diff
@@ -299,3 +299,266 @@ export function useAllSubmissions(limit: number = 20, status?: string) {
+      const totalRejections = stats.totalSubmissions - stats.totalApprovals;
+
+      // Calculate approval rate
+      const approvalRate =
+        stats.totalApprovals + totalRejections > 0
+          ? stats.totalApprovals / (stats.totalApprovals + totalRejections)
+          : 0;
+
+      return {
+        totalSubmissions: stats.totalSubmissions,
+        totalApprovals: stats.totalApprovals,
+        totalRejections,
+        approvalRate,
+      };
+    },
+    enabled: !!web3auth,
+  });
+}
+
+export function useUserActivity(
+  userId: string | number,
+  options?: ActivityQueryOptions,
+) {
+  const params = new URLSearchParams();
+  if (options?.timeRange) params.append("timeRange", options.timeRange);
+  if (options?.feedId) params.append("feedId", options.feedId);
+
+  const queryString = params.toString();
+
+  return useQuery<AggregatedActivityStats>({
+    queryKey: ["user-activity", userId, options],
+    queryFn: async () => {
+      const url = `/api/activity/user/${userId}${queryString ? `?${queryString}` : ""}`;
+      const response = await fetch(url);
+
+      if (!response.ok) {
+        throw new Error("Failed to fetch user activity");
+      }
+
+      const data: ListOfUserActivityStats = await response.json();
+
+      // Calculate statistics from activities
+      const stats = data.activities.reduce(
+        (acc, activity) => {
+          if (activity.type === ActivityType.CONTENT_SUBMISSION) {
+            acc.totalSubmissions++;
+          } else if (activity.type === ActivityType.CONTENT_APPROVAL) {
+            acc.totalApprovals++;
+          }
+          return acc;
+        },
+        { totalSubmissions: 0, totalApprovals: 0 },
+      );
+
+      // Calculate rejections (submissions - approvals)
+      const totalRejections = stats.totalSubmissions - stats.totalApprovals;
+
+      // Calculate approval rate
+      const approvalRate =
+        stats.totalApprovals + totalRejections > 0
+          ? stats.totalApprovals / (stats.totalApprovals + totalRejections)
+          : 0;
+
+      return {
+        totalSubmissions: stats.totalSubmissions,
+        totalApprovals: stats.totalApprovals,
+        totalRejections,
+        approvalRate,
+      };
+    },
+  });
+}
+
+export function useMyCuratedFeeds() {
+  const { web3auth } = useWeb3Auth();
+
+  return useQuery<CuratedFeed[]>({
+    queryKey: ["my-curated-feeds"],
+    queryFn: async () => {
+      if (!web3auth) throw new Error("Web3Auth not initialized");
+      const authResult = await web3auth.authenticateUser();
+
+      const response = await fetch("/api/activity/feeds/curated-by/me", {
+        headers: {
+          Authorization: `Bearer ${authResult.idToken}`,
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error("Failed to fetch curated feeds");
+      }
+
+      return response.json();
+    },
+    enabled: !!web3auth,
+  });
+}
+
+export function useMyApprovedFeeds() {
```

Comment:
Nice -- now this should be used by 

<img width="594" alt="Screenshot 2025-05-15 at 9 06 54 AM" src="https://github.com/user-attachments/assets/74af18d8-d783-4e7b-876f-9da3662fe067" />